pkgs/top-level/lua-packages.nix

```
@@ -59,6 +59,47 @@ rec {
   # a fork of luarocks used to generate nix lua derivations from rockspecs
   luarocks-nix = toLuaModule (callPackage ../development/tools/misc/luarocks/luarocks-nix.nix { });
 
+ lua-pam = callPackage({fetchFromGitHub, linux-pam, openpam}: buildLuaPackage rec {
+    pname = "lua-pam";
+    version = "unstable-2015-07-03";
+    # Needed for `disabled`, overridden in buildLuaPackage
+    name = "${pname}-${version}";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "devurandom";
+      repo = "lua-pam";
+      rev = "3818ee6346a976669d74a5cbc2a83ad2585c5953";
+      hash = "sha256-YlMZ5mM9Ij/9yRmgA0X1ahYVZMUx8Igj5OBvAMskqTg=";
+      fetchSubmodules = true;
+    };
+
+    # The makefile tries to link to `-llua<luaversion>`
+    LUA_LIBS = "-llua";
+
+    buildInputs =
+      (lib.lists.optional stdenv.isLinux linux-pam)
+      ++ (lib.lists.optional stdenv.isDarwin openpam);
```

```suggestion
    buildInputs = lib.optional stdenv.isLinux linux-pam
      ++ lib.optional stdenv.isDarwin openpam;
```
lib.lists is an implementaion detail we shouldn't use. Also the () here are superfluous.